clap_derive/tests/flatten.rs

```diff
@@ -160,3 +162,76 @@ fn update_subcommands_with_flatten() {
     opt.update_from(&["test", "command2", "43"]);
     assert_eq!(Opt::parse_from(&["test", "command2", "43"]), opt);
 }
+
+#[test]
+fn docstrings_ordering_with_multiple_clap() {
+    /// this is the docstring for Flattened
+    #[derive(Clap)]
+    struct Flattened {
+        #[clap(long)]
+        foo: bool,
+    }
+
+    /// this is the docstring for Command
+    #[derive(Clap)]
+    struct Command {
+        #[clap(flatten)]
+        flattened: Flattened,
+    }
+
+    static ISSUE_2527: &str = "clap_derive 
+
+this is the docstring for Command
+
+USAGE:
+    command [FLAGS]
+
+FLAGS:
+        --foo        
+    -h, --help       Prints help information
+    -V, --version    Prints version information
+";
+
+    assert!(utils::compare_output(
+        Command::into_app(),
+        "command --help",
+        ISSUE_2527,
+        false
+    ));
+}
+
+#[test]
+fn docstrings_ordering_with_multiple_clap_partial() {
+    /// this is the docstring for Flattened
+    #[derive(Clap)]
+    struct Flattened {
+        #[clap(long)]
+        foo: bool,
+    }
+
+    #[derive(Clap)]
+    struct Command {
+        #[clap(flatten)]
+        flattened: Flattened,
+    }
```

Comment:
Not sure what should be the behavior for this. Usually people won't just write comments on one side. Fixing this requires quite some effort I think since we need to be able to check whether check whether `augment_clap` is on the root.